test/e2e/vsphere_volume_placement.go

```diff
@@ -17,220 +17,370 @@ limitations under the License.
+			podA                *v1.Pod
+			podB                *v1.Pod
+			testvolumePathsPodA []string
+			testvolumePathsPodB []string
+			podAFiles           []string
+			podBFiles           []string
+		)
+
+		defer func() {
+			By("clean up undeleted pods")
+			framework.DeletePodWithWait(f, c, podA)
+			framework.DeletePodWithWait(f, c, podB)
+			By(fmt.Sprintf("wait for volumes to be detached from the node: %v", node1Name))
+			for _, volumePath := range volumePaths {
+				waitForVSphereDiskToDetach(vsp, volumePath, types.NodeName(node1Name))
+			}
+		}()
+
+		testvolumePathsPodA = append(testvolumePathsPodA, volumePaths[0])
+		// Create another VMDK Volume
+		By("creating another vmdk")
+		volumePath, err := createVSphereVolume(vsp, nil)
+		Expect(err).NotTo(HaveOccurred())
+		volumePaths = append(volumePaths, volumePath)
+		testvolumePathsPodB = append(testvolumePathsPodA, volumePath)
+
+		for index := 0; index < 5; index++ {
+			By(fmt.Sprintf("Creating pod-A on the node: %v with volume: %v", node1Name, testvolumePathsPodA[0]))
+			podA = createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, testvolumePathsPodA)
+
+			By(fmt.Sprintf("Creating pod-B on the node: %v with volume: %v", node1Name, testvolumePathsPodB[0]))
+			podB = createPodWithVolumeAndNodeSelector(c, ns, vsp, node1Name, node1KeyValueLabel, testvolumePathsPodB)
+
+			podAFileName := fmt.Sprintf("/mnt/volume1/podA_%v_%v.txt", ns, index+1)
+			podBFileName := fmt.Sprintf("/mnt/volume1/podB_%v_%v.txt", ns, index+1)
+			podAFiles = append(podAFiles, podAFileName)
+			podBFiles = append(podBFiles, podBFileName)
+
+			// Create empty files on the mounted volumes on the pod to verify volume is writable
+			By("Creating empty file on volume mounted on pod-A")
+			createEmptyFileOnVSphereVolume(ns, podA.Name, podAFileName)
+
+			By("Creating empty file volume mounted on pod-B")
+			createEmptyFileOnVSphereVolume(ns, podB.Name, podBFileName)
+
+			// Verify newly and previously created files present on the volume mounted on the pod
+			By("Verify newly Created file and previously created files present on volume mounted on pod-A")
+			verifyFilesExistOnVSphereVolume(ns, podA.Name, podAFiles)
+			By("Verify newly Created file and previously created files present on volume mounted on pod-B")
+			verifyFilesExistOnVSphereVolume(ns, podB.Name, podBFiles)
+
+			By("Deleting pod-A")
+			framework.DeletePodWithWait(f, c, podA)
+			By("Deleting pod-B")
+			framework.DeletePodWithWait(f, c, podB)
+		}
 	})
 })
 
-func createPodWithVolumeAndNodeSelector(client clientset.Interface, namespace string, vsp *vsphere.VSphere, nodeName string, nodeKeyValueLabel map[string]string, volumePath string) *v1.Pod {
+func testSetupVolumePlacement(client clientset.Interface, namespace string) (node1Name string, node1KeyValueLabel map[string]string, node2Name string, node2KeyValueLabel map[string]string) {
+	nodes := framework.GetReadySchedulableNodesOrDie(client)
+	if len(nodes.Items) < 2 {
```

Comment:
`e2e.go -> setupProviderConfig() `does not read test config for vsphere provider. vsphere tests are executed as conformance tests without setting --provider flag

```
export KUBERNETES_CONFORMANCE_PROVIDER="vsphere"
export KUBERNETES_CONFORMANCE_TEST=Y
```